File: data/src/main/java/home/oleg/placesnearme/data/repositories/DistanceRepositoryImpl.java
package home.oleg.placesnearme.data.repositories;

import android.location.Location;

import home.oleg.placenearme.models.LatLng;
import home.oleg.placenearme.repositories.DistanceRepository;

/**
 * Created by Oleg Sheliakin on 09.10.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
public class DistanceRepositoryImpl implements DistanceRepository {

    @Override
    public double evaluate(LatLng from, LatLng to) {
        Location locationFrom = create(from);
        Location locationTo = create(to);
        return locationFrom.distanceTo(locationTo);
    }

    private Location create(LatLng latLng) {
        Location location = new Location("");
        location.setLatitude(latLng.getLat());
        location.setLongitude(latLng.getLng());
        return location;
    }
}

File: core-data/src/main/java/home/oleg/placesnearme/coredata/repositories/DistanceRepositoryImpl.kt
package home.oleg.placesnearme.coredata.repositories

import android.location.Location

import javax.inject.Inject

import home.oleg.placesnearme.coredomain.models.LatLng
import home.oleg.placesnearme.coredomain.repositories.DistanceRepository

/**
 * Created by Oleg Sheliakin on 09.10.2018.
 * Contact me by email - olegsheliakin@gmail.com
 */
class DistanceRepositoryImpl @Inject constructor() : DistanceRepository {

    override fun evaluate(from: LatLng, to: LatLng): Double {
        val locationFrom = create(from)
        val locationTo = create(to)
        return locationFrom.distanceTo(locationTo).toDouble()
    }

    private fun create(latLng: LatLng): Location {
        return Location("").apply {
            latitude = latLng.lat
            longitude = latLng.lng
        }
    }
}
